### DISCUSSION TITLE
can i define multiple reframe db to implement fsm?

### DISCUSSION THREAD

**Discussion by maochaokuo** [day8/re-frame#736] [Q&A]
hi there, assume I have two textboxes, very important and complicated for both of them. can i both create one  reframe db for each of them? because i need one reframe db to implement one fsm for each textbox. therefore, is it possible for me to create as many reframe db as possible, for each tiny small machine or engine?

I am sorry if i post the wrong place, because i see no community for reframe. If reframe does not support multiple db, then i will have to implement fsm myself. thanks a lot for any help

**Comment by superstructor** [comment]
Multi-instanced re-frame is a feature of a future release. Each instance would be a 'frame'. Each 'frame' consisting of app-db, the registrar atoms (e.g. event handlers, subs etc) and so forth. Then each 'frame' would be passed down the component tree using [React Contexts](https://reactjs.org/docs/context.html).

This feature has been designed but not yet implemented.

At the moment if you want to keep things separate you need to do it manually in your db structure / event handlers etc.
